packages/db/lib/migrator.mjs

```diff
@@ -45,14 +56,46 @@ async function execute (logger, args, config) {
       'migration-finished',
       (migration) => logger.debug(`completed ${basename(migration.filename)}`)
     )
+    const migrations = await postgrator.getMigrations()
+    const current = await postgrator.getDatabaseVersion()
+
+    if (args.up) {
+      const nextUp = migrations.filter(x => x.version > current)
+        .filter(x => x.action === 'do')
+        .sort((a, b) => a.version - b.version)
+        .map(extractVersionFromMigration)[0]
```

Comment:
I recommend using `flatMap` instead of `.filter().map()` to reduce traversing the array 2 times.

For example:

```javascript
array.flatMap(v => isFalsy(v) ? [] : [extractVersionFromMigration(v)])
```